File: 13. BST 5.java
import java.util.*;

class Node {
    int data;
    Node left, right;
    Node(int data) {
        this.data = data;
        left = right = null;
    }
}

class BinaryTree {
    int idx = 0;
    public Node buildTree(List<Integer> arr) 
    {
        if (idx >= arr.size() || arr.get(idx) == -1) 
        {
            idx++;
            return null;
        }

        Node root = new Node(arr.get(idx++));
        root.left = buildTree(arr);
        root.right = buildTree(arr);

        return root;
    }


    public void levelOrder(Node root) {
        if (root == null) return;

        Queue<Node> q = new LinkedList<>();
        q.add(root);

        while (!q.isEmpty()) {
            Node curr = q.poll();
            System.out.print(curr.data + " ");

            if (curr.left != null) q.add(curr.left);
            if (curr.right != null) q.add(curr.right);
        }
    }
}

public class Main {
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        ArrayList<Integer> list = new ArrayList<>();

        while (sc.hasNextInt()) 
        {
            list.add(sc.nextInt());
        }
        BinaryTree t=new BinaryTree();
        Node root = t.buildTree(list);
        t.levelOrder(root);
    }
}

File: 13. BST 5.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None
class BinaryTree:
    def __init__(self):
        self.idx = 0
    def build_tree(self, arr):
        if self.idx >= len(arr):
            return None
        if arr[self.idx] == -1:
            self.idx+=1
            return None
        root = Node(arr[self.idx])
        self.idx += 1
        root.left = self.build_tree(arr)
        root.right = self.build_tree(arr)
        return root

    def level_order(self, root):
        q = []
        q.append(root)
        result = []
        while q:
            node = q.pop(0)
            result.append((node.data))
            if node.left:
                q.append(node.left)
            if node.right:
                q.append(node.right)
        print(*result)

import sys
arr = []
arr = list(map(int, sys.stdin.read().split()))
bt = BinaryTree()
root = bt.build_tree(arr)
bt.level_order(root)
